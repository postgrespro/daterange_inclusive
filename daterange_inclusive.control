# daterange_inclusive extension
comment = 'daterange with inclusive upper bound'
default_version = '1.0'
module_pathname = '$libdir/daterange_inclusive'
relocatable = true
trusted = true

