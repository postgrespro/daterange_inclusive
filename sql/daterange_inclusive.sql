CREATE EXTENSION daterange_inclusive;
SELECT daterange_inclusive('2024-01-01' , '2024-06-01', '[]');
SELECT daterange_inclusive('2024-01-01' , '2024-06-01', '[)');
SELECT '[2025-01-01,2025-02-02]'::daterange::daterange_inclusive;
SELECT '[2025-01-01,2025-02-02)'::daterange::daterange_inclusive;
SELECT '[2025-01-01,2025-02-02]'::daterange_inclusive::daterange;
SELECT '[2025-01-01,2025-02-02)'::daterange_inclusive::daterange;

CREATE TABLE x (y daterange_inclusive);
INSERT INTO x SELECT daterange_inclusive(d::date, (d + '1month'::interval)::date, '[]') FROM generate_series('2015-01-01'::date,'2025-02-02', '1day') d;
INSERT INTO x SELECT daterange_inclusive(d::date, NULL::date, '[)') FROM generate_series('2015-01-01'::date,'2025-02-02', '1day') d;
SELECT COUNT(*) FROM x;
CREATE INDEX ON x USING gist (y);
INSERT INTO x SELECT '[2024-01-01,)'::daterange::daterange_inclusive;

DROP EXTENSION daterange_inclusive CASCADE;

