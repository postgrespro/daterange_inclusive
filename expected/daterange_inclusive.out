CREATE EXTENSION daterange_inclusive;
SELECT daterange_inclusive('2024-01-01' , '2024-06-01', '[]');
   daterange_inclusive   
-------------------------
 [01-01-2024,06-01-2024]
(1 row)

SELECT daterange_inclusive('2024-01-01' , '2024-06-01', '[)');
   daterange_inclusive   
-------------------------
 [01-01-2024,05-31-2024]
(1 row)

SELECT '[2025-01-01,2025-02-02]'::daterange::daterange_inclusive;
   daterange_inclusive   
-------------------------
 [01-01-2025,02-02-2025]
(1 row)

SELECT '[2025-01-01,2025-02-02)'::daterange::daterange_inclusive;
   daterange_inclusive   
-------------------------
 [01-01-2025,02-01-2025]
(1 row)

SELECT '[2025-01-01,2025-02-02]'::daterange_inclusive::daterange;
        daterange        
-------------------------
 [01-01-2025,02-03-2025)
(1 row)

SELECT '[2025-01-01,2025-02-02)'::daterange_inclusive::daterange;
        daterange        
-------------------------
 [01-01-2025,02-02-2025)
(1 row)

CREATE TABLE x (y daterange_inclusive);
INSERT INTO x SELECT daterange_inclusive(d::date, (d + '1month'::interval)::date, '[]') FROM generate_series('2015-01-01'::date,'2025-02-02', '1day') d;
INSERT INTO x SELECT daterange_inclusive(d::date, NULL::date, '[)') FROM generate_series('2015-01-01'::date,'2025-02-02', '1day') d;
SELECT COUNT(*) FROM x;
 count 
-------
  7372
(1 row)

CREATE INDEX ON x USING gist (y);
INSERT INTO x SELECT '[2024-01-01,)'::daterange::daterange_inclusive;
DROP EXTENSION daterange_inclusive CASCADE;
NOTICE:  drop cascades to column y of table x
